tests/network-tests/src/fixtures/forum/RemarkModeratePostsFixture.ts

```diff
@@ -0,0 +1,96 @@
+import { RemarkMetadataAction } from '@joystream/metadata-protobuf'
+import { Api } from '../../Api'
+import { QueryNodeApi } from '../../QueryNodeApi'
+import { EventDetails } from '../../types'
+import { Utils } from '../../utils'
+import { ForumPostFieldsFragment, PostModeratedEventFieldsFragment } from '../../graphql/generated/queries'
+import { WithForumWorkersFixture } from './WithForumWorkersFixture'
+
+export type RemarkPostModerationInput = {
+  postId: ForumPostId
+  rationale: string
+  asWorker?: WorkerId
+  expectFailure?: boolean
+}
+
+export class RemarkModeratePostsFixture extends WithForumWorkersFixture {
+  protected moderations: RemarkPostModerationInput[]
+
+  public constructor(api: Api, query: QueryNodeApi, moderations: RemarkPostModerationInput[]) {
+    super(api, query)
+    this.moderations = moderations
+  }
+
+  protected async getSignerAccountOrAccounts(): Promise<string[]> {
+    return this.getSignersFromInput(this.moderations)
+  }
+
+  protected async getExtrinsics(): Promise<SubmittableExtrinsic<'promise'>[]> {
+    return this.moderations.map((u) => {
+      const metadata = Utils.metadataToBytes(RemarkMetadataAction, {
+        moderatePost: { postId: Long.fromString(String(u.postId)), rationale: u.rationale },
+      })
+      return u.asWorker
+        ? this.api.tx.forumWorkingGroup.workerRemark(u.asWorker, metadata)
+        : this.api.tx.forumWorkingGroup.leadRemark(metadata)
+    })
+  }
+
+  protected async getEventFromResult(result: ISubmittableResult): Promise<EventDetails> {
+    if (this.api.findEvent(result, 'forumWorkingGroup', 'WorkerRemarked')) {
+      return this.api.getEventDetails(result, 'forumWorkingGroup', 'WorkerRemarked')
+    } else {
+      return this.api.getEventDetails(result, 'forumWorkingGroup', 'LeadRemarked')
+    }
+  }
+
+  protected assertQueriedPostsAreValid(
+    qPosts: ForumPostFieldsFragment[],
+    qEvents: PostModeratedEventFieldsFragment[]
+  ): void {
+    const moderatedSuccessfully = this.moderations.filter((m) => !m.expectFailure).length
+    assert.equal(qEvents.length, moderatedSuccessfully, 'Too many posts were moderated')
+
+    this.events.map((e, i) => {
+      const moderation = this.moderations[i]
+      const qPost = qPosts.find((p) => p.id === moderation.postId.toString())
+
+      Utils.assert(qPost, 'Query node: Post not found')
+
+      if (moderation.expectFailure) return
+
+      const qEvent = this.findMatchingQueryNodeEvent(e, qEvents)
+      Utils.assert(qPost.status.__typename === 'PostStatusModerated', 'Invalid post status')
+      Utils.assert(qPost.status.postModeratedEvent, 'Query node: Missing PostModeratedEvent ref')
+      assert.equal(qPost.status.postModeratedEvent.id, qEvent.id)
+      assert.equal(qPost.isVisible, false)
+    })
+  }
+
+  protected assertQueryNodeEventIsValid(qEvent: PostModeratedEventFieldsFragment, i: number): void {
+    const { postId, asWorker, rationale } = this.moderations[i]
```

Comment:
Since there will be no `qEvent` if `expectFailure === true`, I think this should be:
```suggestion
    const { postId, asWorker, rationale } = this.moderations[i].filter((m) => !m.expectFailure)
```

Reply:
I think I did this at first but then (if I remember correctly) it would failed because `i` comes from [tests/network-tests/src/Fixture.ts](https://github.com/Joystream/joystream/pull/4317/files/b04b048f623422d36171d54bbf5da00d7a80b9aa#diff-a69963794564bd61b084b85a260e2642dc7faa6a6d6bee805dfbb6e426306e64) which includes `expectFailure` but then skip them. so e.g: if `this.moderations[1]` is `expectFailure` then `assertQueryNodeEventIsValid` will be called with `i: 0, 2, ...` but not `1`

Reply:
You're right, my bad :+1: